test/integration/auth.test.js

```diff
@@ -254,4 +256,288 @@ describe("auth", function () {
     expect(res.headers["set-cookie"][1]).to.include(`Domain=${rdsUiUrl.hostname}`);
     expect(res.headers["set-cookie"][1]).to.include("SameSite=Lax");
   });
+
+  it("should redirect to the correct URL and update user email when GitHub API returns primary email", async function () {
+    const rdsUrl = new URL(config.get("services.rdsUi.baseUrl")).href;
+    const fakeEmails = [
+      { primary: true, email: "primary@example.com" },
+      { primary: false, email: "secondary@example.com" },
+    ];
+    const fetchStub = sinon.stub(global, "fetch").resolves(new Response(JSON.stringify(fakeEmails)));
+
+    sinon.stub(passport, "authenticate").callsFake((strategy, options, callback) => {
+      callback(null, "accessToken", {
+        username: "github-user",
+        displayName: "GitHub User",
+        _json: { email: null, created_at: "2022-01-01" },
+        id: 12345,
+      });
+      return (req, res, next) => {};
+    });
+
+    const res = await chai
+      .request(app)
+      .get(`/auth/github/callback`)
+      .query({ code: "codeReturnedByGithub", state: rdsUrl })
+      .redirects(0);
+    expect(res).to.have.status(302);
+
+    const fetchArgs = fetchStub.getCall(0).args;
+    expect(fetchArgs[0]).to.equal("https://api.github.com/user/emails");
+    expect(fetchArgs[1].headers.Authorization).to.equal("token accessToken");
+  });
+
+  it("should return google call back URL", async function () {
+    const googleOauthURL = generateGoogleAuthRedirectUrl({});
+    const res = await chai.request(app).get("/auth/google/login?dev=true").redirects(0);
+    expect(res).to.have.status(302);
+    expect(res.headers.location).to.equal(googleOauthURL);
+  });
+
+  it("should return google call back URL with redirectUrl", async function () {
+    const RDS_MEMBERS_SITE_URL = "https://members.realdevsquad.com";
+    const googleOauthURL = generateGoogleAuthRedirectUrl({ state: RDS_MEMBERS_SITE_URL });
+    const res = await chai
+      .request(app)
+      .get("/auth/google/login?dev=true")
+      .query({ redirectURL: RDS_MEMBERS_SITE_URL })
+      .redirects(0);
+    expect(res).to.have.status(302);
+    expect(res.headers.location).to.equal(googleOauthURL);
+  });
+
+  it("should redirect the google user to new sign up flow if they are have incomplete user details true", async function () {
+    const redirectURL = "https://my.realdevsquad.com/new-signup";
+    stubPassportAuthenticate(googleUserInfo[0]);
+
+    const res = await chai
+      .request(app)
+      .get("/auth/google/callback")
```

Comment:
Shouldn't this also be behind a flag?

Reply:
RE: 
When Google redirects to the callback URL, it usually appends its own parameters (such as code, state, etc.) but will not include any query parameters like dev=true. and if a user directly visits this url he will not be able to authenticate and receive 401 error.

Reply:
is there no way to have a feature flag similar behavior to achieve here?

Reply:
Google and other oauth providers only allow state, redirect uri, etc parameters to be passed rest are ignored. [Mentioned here The Google authorization server supports the following query string parameters for web server applications:](https://developers.google.com/identity/protocols/oauth2/web-server#creatingclient).

Reply:
Moreover this route cannot be accessed directly by any user if accesed directly it will always give 401.